src/dotnet/ShellShim/AppHostShimMaker.cs

```diff
@@ -41,12 +42,29 @@ public void CreateApphostShellShim(FilePath entryPoint, FilePath shimPath)
             }
 
             var appHostDestinationFilePath = Path.GetFullPath(shimPath.Value);
-            var appBinaryFilePath = Path.GetRelativePath(Path.GetDirectoryName(appHostDestinationFilePath), Path.GetFullPath(entryPoint.Value));
+            string entryPointFullPath = Path.GetFullPath(entryPoint.Value);
+            var appBinaryFilePath = Path.GetRelativePath(Path.GetDirectoryName(appHostDestinationFilePath), entryPointFullPath);
 
-            EmbedAppNameInHost.EmbedAndReturnModifiedAppHostPath(
-                appHostSourceFilePath: appHostSourcePath,
-                appHostDestinationFilePath: appHostDestinationFilePath,
-                appBinaryFilePath: appBinaryFilePath);
+
+            if (ResourceUpdater.IsSupportedOS())
+            {
+                var windowsGraphicalUserInterfaceBit = BinaryUtils.GetWindowsGraphicalUserInterfaceBit(entryPointFullPath);
```

Comment:
Side issue for core-setup, this isn't a very well named API. It returns the subsystem, not a single bit for GUI or not GUI. BinaryUtils is also very unfortunate name for public API.

FYI, @swaroop-sridhar